File: src/test/java/com/serverest/client/CartClient.java
package com.serverest.client;

import com.serverest.config.Specs;
import com.serverest.constants.Routes;
import io.restassured.response.Response;

import java.util.List;
import java.util.Map;

import static io.restassured.RestAssured.given;

public class CartClient {

    public Response create(String productId, int quantity, String token) {
        Map<String, Object> cartPayload = Map.of(
                "produtos", List.of(Map.of("idProduto", productId, "quantidade", quantity))
        );
        return given()
                .spec(Specs.requestSpec())
                .header("Authorization", token)
                .body(cartPayload)
                .when()
                .post(Routes.CARTS);
    }

    public Response cancel(String token) {
        return given()
                .spec(Specs.requestSpec())
                .header("Authorization", token)
                .when()
                .delete(Routes.CARTS_CANCEL);
    }

    public Response conclude(String token) {
        return given()
                .spec(Specs.requestSpec())
                .header("Authorization", token)
                .when()
                .delete(Routes.CARTS_CONCLUDE);
    }
}

File: utils/clients/cartClient.ts
import type { APIRequestContext } from "@playwright/test";
import { apiCall } from "./baseApi";

export class CartClient {
  constructor(private ctx: APIRequestContext) {}

  async create(productId: string, quantity: number, authToken: string) {
    const payload = { produtos: [{ idProduto: productId, quantidade: quantity }] };
    return apiCall<{ _id: string; message: string }>(this.ctx, "post", "/carrinhos", {
      data: payload,
      headers: { Authorization: authToken }
    });
  }

  async conclude(authToken: string) {
    return apiCall<{ message: string }>(this.ctx, "delete", "/carrinhos/concluir-compra", {
      headers: { Authorization: authToken }
    });
  }

  async cancel(authToken: string) {
    return apiCall<{ message: string }>(this.ctx, "delete", "/carrinhos/cancelar-compra", {
      headers: { Authorization: authToken }
    });
  }
}
